### PR TITLE
Change chat appearance

### PR DESCRIPTION
**Pull request by ManuViola77**
This PR is to improve chat's appearance since the last PR was only functional but not nice to the eye. 

It also adds the option to get older chats by pressing the `load earlier messages` button (it fetches 10 each time since that is how backend returns pages with 10 elements).

There was a complication about adding new messages (now I have more than 10) and then fetching older messages. To handle this I do 2 things:

1. I don't keep track of the last page I got, I calculate which one I have to get now with this formula: `Math.floor(messages.length / CHAT_PAGE_COUNT) + 1` having a configuration `CHAT_PAGE_COUNT=10`. If backend changes this configuration has to change too.

2.  When I do step one, I could have repeated messages, for example: 
-  I had 10 messages
- then I send or receive 2 new messages so I have 12
- when I fetch older messages I have to get page 2, but the last 2 messages would be the same as the first 2 I already have now (that difference between 12 and 10, now the first 2 are part of the next page, in this case, page 2).
To address this I have to check, when I fetch a page different than page 1, if it's already in the collection don't add it again.

I leave an example where I add texts in numerical and alphabetical order so it's easy to check there is no message repeated or missing.

EDIT:  I saw in the gif that one user's avatar wasn't showing because is empty, now in these cases, I set default avatar: 

image

### CODE REVIEW

File: src/screens/ChatScreen/index.js [hunk 1]
```diff
@@ -57,19 +78,41 @@ const ChatScreen = () => {
...
-        )}
-        invertibleScrollViewProps={{ overScrollMode: 'never' }}
         alignTop
+        alwaysShowSend
         bottomOffset={0}
+        /*
+        this explains why I set extraData:
+        https://www.gitmemory.com/issue/FaridSafi/react-native-gifted-chat/1182/506336547
+        */
+        extraData={{ isLoadingEarlier }}
```
Comment (peter7z) [inline src/screens/ChatScreen/index.js]: Extra pair of parenthesis here.

File: src/screens/ChatScreen/index.js [hunk 2]
```diff
@@ -42,13 +48,28 @@ const ChatScreen = () => {
 
   let messages = useMemo(() => {
```
Comment (peter7z) [inline src/screens/ChatScreen/index.js]: Use const here
```js
const messages = useMemo(() =>
  messagesSession?
    .map(({
      // message things here...
    }) => ({
      // message things to return here
    }).reverse()
);
```

File: src/reducers/chatReducer.js [hunk 3]
```diff
@@ -17,8 +21,15 @@ const chatReducer = {
     store.matches = payload;
   },
 
-  [getMessagesSuccess]: (store, { payload }) => {
-    store.messages = payload;
+  [getMessagesSuccess]: (store, { payload: { messages, page } }) => {
+    if (page !== 1) {
+      const newMessages = messages.filter(({ id }) => {
+        return !isInCollection(store.messages, id);
+      });
```
Comment (peter7z) [inline src/reducers/chatReducer.js]:
```suggestion
      const newMessages = messages.filter(({ id }) => !isInCollection(store.messages, id);
```

File: src/reducers/chatReducer.js [hunk 4]
```diff
@@ -8,6 +8,10 @@ import {
 
 const initialState = { matches: [] };
 
+const isInCollection = (array, id) => {
+  return array.find(({ id: elementId }) => elementId === id);
+};
```
Comment (peter7z) [inline src/reducers/chatReducer.js]:
```suggestion
const isInCollection = (array, id) => array.find(({ id: elementId }) => elementId === id);
```

File: src/screens/ChatScreen/index.js (hunk 1 above)
Comment (ManuViola77) [reply src/screens/ChatScreen/index.js]: It expects an object there. 

image

**Pull request merged**